samples/sgs.py

```diff
@@ -0,0 +1,333 @@
+    )
+
+    output_file = []
+    header_added = False
+    for line in file_lines:
+        if IMPORTS_FILL.search(line):
+            output_file.append(import_lines)
+        elif match := INGREDIENT_FILL.search(line):
+            output_file.append(ingredients[match.group(1)].text)
+        elif REGION_START.search(line):
+            output_file.append(REGION_START.sub("# [START \\1]", line))
+        elif REGION_END.search(line):
+            output_file.append(REGION_END.sub("# [END \\1]", line))
+        else:
+            output_file.append(line)
+            continue
+        if not header_added:
+            end = output_file[-1]
+            output_file[-1] = ""
+            output_file.append(HEADER)
+            output_file.append("")
+            output_file.append(end)
+            header_added = True
+
+    if output_file and not output_file[-1].endswith("\n"):
+        output_file.append("")
+
+    return os.linesep.join(output_file)
+
+
+def save_rendered_recipe(
+    recipe_path: Path,
+    rendered_recipe: str,
+    output_dir: Path = DEFAULT_OUTPUT_PATH,
+    recipes_path: Path = RECIPES_PATH,
+) -> Path:
+    output_dir.mkdir(exist_ok=True)
+
+    output_path = output_dir / recipe_path.relative_to(recipes_path)
+    output_path.parent.mkdir(exist_ok=True)
+
+    with output_path.open(mode="w") as out_file:
+        out_file.write(rendered_recipe)
+
+    subprocess.run(
+        ["black", str(output_path)],
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.DEVNULL,
+    )
+    return output_path
+
+
+def generate(
+    args: argparse.Namespace,
+    ingredients_path: Path = INGREDIENTS_PATH,
+    recipes_path: Path = RECIPES_PATH,
+):
+    ingredients = load_ingredients(ingredients_path)
+    recipes = load_recipes(recipes_path)
+
+    updated_paths = set()
+
+    for path, recipe in recipes.items():
+        rendered = render_recipe(recipe, ingredients)
+        out = save_rendered_recipe(path, rendered, output_dir=Path(args.output_dir))
+        updated_paths.add(str(out))
+
+    print("Generated files:")
+    for file in sorted(updated_paths):
+        print(f" - {repr(file)}")
+
+    all_files = glob.glob(f"{args.output_dir}/**", recursive=True)
+    unknown_files = set()
+    for file in all_files:
+        if file in updated_paths:
+            continue
+        if any(pattern.match(file) for pattern in IGNORED_OUTPUT_FILES):
+            continue
+        pfile = Path(file)
+        if pfile.is_dir() and pfile.iterdir():
+            # Don't report non-empty dirs.
+            continue
+        unknown_files.add(file)
+
+    if unknown_files:
+        print("Found following unknown files: ")
+        for file in sorted(unknown_files):
+            print(f" - {repr(file)}")
+
+
+def verify(args: argparse.Namespace):
+    # TODO: Needs to check if the files are up to date. Will be used to auto-check every commit.
+    pass
+
+
+def parse_arguments():
```

Comment:
Let's add to the README file input params doc and example of usage.
Example: https://github.com/GoogleCloudPlatform/compute-migration-scripts/tree/main/instances-to-stateful-mig#usage

Reply:
That's really pretty :P I'll make something similar I guess.